src/reanimated2/initializers.ts

```diff
@@ -0,0 +1,150 @@
+import NativeReanimatedModule from './NativeReanimated';
+import { runOnUI, runOnJS } from './threads';
+
+function callGuard<T extends Array<any>, U>(
+  fn: (...args: T) => U,
+  ...args: T
+): void {
+  'worklet';
+  try {
+    fn(...args);
+  } catch (e) {
+    if (global.ErrorUtils) {
+      global.ErrorUtils.reportFatalError(e);
+    } else {
+      throw e;
+    }
+  }
```

Comment:
What about using `try { ... } catch { ... }` only in development mode (`__DEV__`)?